File: SecondYear/sauban123_MohdSauban_2226csit1130_2/Week_3/BinarySearch.cpp
int search(vector<int> &nums, int target) {
    // Write your code here.
    int l=0;
    int h =nums.size()-1;
    int mid = (l+h)/2;
    while(l<=h){
        if(nums[mid] == target){
            return mid;
        }
        else if(nums[mid]>target){
            h= mid-1;
        }
        else {
            l=mid+1;
        }
        mid = (l+h)/2;
    }
    return -1;
}

File: SecondYear/aayushker_AayushkerSingh_2226CSE(AI)1010_2/Week_3/Binary_Search.py
def search(nums: [int], target: int):
    # write your code logic !!
    l, r = 0, len(arr) - 1

    while l <= r:
        mid = (l + r) // 2

        if arr[mid] == target:
            return mid

        elif arr[mid] < target:
            l = mid + 1

        else:
            r = mid - 1 

    return -1

n= int (input())
arr = list(map(int,input().strip().split()))[:n]
target =int (input())
print (search(arr, target))